### PR TITLE
Added warning when adding a column to a table with different length

### PR DESCRIPTION
**Pull request by MoritzNeuberger**
I have been working on implementing warnings thrown when adding a column to a table with a different length. This is an implementation of issue 27 [[link](https://github.com/legend-exp/legend-pydataobj/issues/27)].

The specific changes:
- I changed the behavior for an empty table. This was necessary to avoid false warnings. Specifically, I changed the default value for `self.size` from `1024` to `None` in the case a table is created with no `size` or `col_dict` argument. If a column is added to a table with `self.size` set to `None`, `self.size` will be set to `len(obj)`. 
- If a column is added to a table with `self.size` unequal to `None` or the length of the new column, a warning is issued. This warning informs the user about the behavior of `add_column` and the possible loss of data. 
- Modified the tests to not throw a warning.

Question: Is `None` a valid value for `self.size`? Should we instead use 0?

What are your thoughts on this approach?

### CODE REVIEW

File: src/lgdo/types/table.py [hunk 1]
```diff
@@ -140,8 +140,19 @@ def add_field(
 
         super().add_field(name, obj)
 
+        if not self.size:
+            self.size = len(obj)
+
         # check / update sizes
         if self.size != len(obj):
+            warn(
```
Comment (gipert) [inline src/lgdo/types/table.py:148]: I think this should be a `logging.warning()`.

File: src/lgdo/types/table.py [hunk 2]
```diff
@@ -140,8 +140,19 @@ def add_field(
 
         super().add_field(name, obj)
 
+        if not self.size:
```
Comment (gipert) [inline src/lgdo/types/table.py:143]: This should be:
```suggestion
        if self.size is not None:
```

**Review by gipert — CHANGES_REQUESTED** [review]
I like this approach of setting `size=None`!

@jasondet thoughts?

File: src/lgdo/types/table.py (hunk 1 above)
Comment (MoritzNeuberger) [reply src/lgdo/types/table.py:148]: `warn()` was also used below in `get_dataframe()`. But we can also replace that one with the logging approach.

File: src/lgdo/types/table.py (hunk 2 above)
Comment (MoritzNeuberger) [reply src/lgdo/types/table.py]: Implemented it as `if self.size is None:` (w/o the `not`)

File: src/lgdo/types/table.py (hunk 1 above)
Comment (gipert) [reply src/lgdo/types/table.py:148]: Yeah I am not sure about the best thing to do in this case see also discussion here:

https://stackoverflow.com/questions/9595009/warnings-warn-vs-logging-warning

You might be right that `warnings.warn()` is better here.

File: src/lgdo/types/table.py (hunk 2 above)
Comment (gipert) [reply src/lgdo/types/table.py]: yes sorry

**Review by gipert — APPROVED** [review]

**Comment by gipert** [comment]
@jasondet can you bless this?

**Pull request merged**